napi-inl.h

```diff
@@ -1632,25 +1632,28 @@ inline Reference<T>::~Reference() {
 }
 
 template <typename T>
-inline Reference<T>::Reference(Reference<T>&& other) {
-  _env = other._env;
-  _ref = other._ref;
+inline Reference<T>::Reference(Reference<T>&& other)
+  : _env(other._env), _ref(other._ref), _suppressDestruct(other._suppressDestruct) {
   other._env = nullptr;
   other._ref = nullptr;
+  other._suppressDestruct = false;
 }
 
 template <typename T>
 inline Reference<T>& Reference<T>::operator =(Reference<T>&& other) {
   Reset();
   _env = other._env;
   _ref = other._ref;
+  _suppressDestruct = other._suppressDestruct;
   other._env = nullptr;
   other._ref = nullptr;
+  other._suppressDestruct = false;
   return *this;
 }
 
 template <typename T>
-inline Reference<T>::Reference(const Reference<T>& other) {
+inline Reference<T>::Reference(const Reference<T>& other) 
+  : _env(other._env), _ref(nullptr), _suppressDestruct(other._suppressDestruct) {
```

Comment:
The purpose of that is explained in the header comment:
```C++
    // Call this on a reference that is declared as static data, to prevent its destructor
    // from running at program shutdown time, which would attempt to reset the reference when
    // the environment is no longer valid.
    void SuppressDestruct();
```